cmd/cmd_controller.js

```diff
@@ -205,22 +205,25 @@ class EmbarkController {
         engine.startService("processManager");
         engine.startService("libraryManager");
         engine.startService("codeRunner");
-        engine.startService("web3");
         if (!options.onlyCompile) {
+          engine.startService("web3");
+          engine.startService("deployment", {onlyCompile: options.onlyCompile});
```

Comment:
Actually, we can drop `onlyCompile` here as internally, the default is `false` anyways.